### PR TITLE
Correct dates in teacher app

### PR DESCRIPTION
**Pull request by megcrenshaw**
The years on the teacher application form were coming in for this current school year (2021-2022), rather than the future one (2022-2023). The red in the error messages are inconsistent (PR coming ...), but the year should be correct.

Old version:
Section 2 Choose Your Program

New:
image

## Links

Many JS strings to be automatically generated are JSON-encoded representations of an object or array. The year is *not* one of those. I used https://gist.github.com/ascendbruce/7070951 as a starting point to convert the year string. If you have a different strategy that's simpler, please share!

- spec: [content doc](https://docs.google.com/document/d/1rshrcpg7J45BfrgiEl6ALwysDeQiwLQcAErFthocJug/edit)
- bug bash doc: [doc](https://docs.google.com/document/d/1Nu30LM1QRFdnXjmbukjdsERBrYGiDx70hBhPTmLI_d0/edit#)

## Testing story

I didn't see a testing file for generateSharedConstants. I manually tested these (details below), but I'm all ears if there are other suggested testing strategies.

To see the years coming through:
1) The only zip code offering a CSA workshop is 98103 (Washington). To get the error message on page 2, choose a school with a different zip code.
2) On page 2, you’ll see an error message if the total number of course hours is less than 100.
3) On page 2, if you select that you are either not plan to teach this course this year, or if you’re not sure if you’ll teach this course, you should get an error message.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/script/generateSharedConstants.rb [hunk 1]
```diff
@@ -38,9 +38,15 @@ def generate_shared_js_file(content, path)
...
-  hash_or_array = hash_or_array.deep_transform_keys {|k| k.to_s.camelize(:lower)} if transform_keys && hash_or_array.is_a?(Hash)
-  "export const #{shared_const_name.downcase.camelize} = #{JSON.pretty_generate(hash_or_array)};"
+  begin
+    hash_or_array = parse_raw(raw)
+    hash_or_array = hash_or_array.deep_transform_keys {|k| k.to_s.camelize(:lower)} if transform_keys && hash_or_array.is_a?(Hash)
+    "export const #{shared_const_name.downcase.camelize} = #{JSON.pretty_generate(hash_or_array)};"
+  rescue JSON::ParserError
+    if raw.is_a?(String)
+      "export const #{shared_const_name.downcase.camelize} = '#{raw}';"
+    end
```
Comment (tim-dot-org) [inline apps/script/generateSharedConstants.rb:50]: maybe, 
`else rethrow the error`

**Review by tim-dot-org — COMMENTED** [review]
in the screenshot, the red text has the old year still, but it looks like it should be updated from the code change, is that just an old screenshot?

**Comment by megcrenshaw** [comment]
> in the screenshot, the red text has the old year still, but it looks like it should be updated from the code change, is that just an old screenshot?

Oh yes ... old screenshot. Will update to include the new one too

**Review by tim-dot-org — APPROVED** [review]
looks great!

**Pull request merged**